parser/src/error.rs

```diff
@@ -54,6 +56,8 @@ pub enum EvalError {
 /// The error for when it required another size of the matrix
 #[derive(Debug, Snafu)]
 pub enum IncompatibleMatrixSizes {
+    // TODO I don't like how we say that something is "expected" here. We can't say
```

Comment:
I compleatly agree. maybe it should just be another type of error where its too small or to big? Like you did down below.